packages/datadog-instrumentations/src/pug.js

```diff
@@ -0,0 +1,24 @@
+'use strict'
+
+const shimmer = require('../../datadog-shimmer')
+const { channel, addHook } = require('./helpers/instrument')
+
+const pugReadCh = channel('datadog:pug:compile:start')
+
+function wrapCompile (compile) {
+  return function wrappedCompile (source) {
+    if (pugReadCh.hasSubscribers) {
+      pugReadCh.publish({ source })
+    }
+
+    return compile.apply(this, arguments)
+  }
+}
+
+addHook({ name: 'pug', versions: ['>=2.0.4'] }, compiler => {
+  shimmer.wrap(compiler, 'compile', wrapCompile)
+  shimmer.wrap(compiler, 'compileClient', wrapCompile)
```

Comment:
Since `compileClient` [calls internally](https://github.com/pugjs/pug/blob/master/packages/pug/lib/index.js#L363) `compileClientWithDependenciesTracked`, and the latter is also instrumented, are we analyzing twice for SSTi when `compileClient` is called?

Reply:
Done